src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,22 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor(popupSelector, handleSubmit) {
+    super(popupSelector);
+    this._handleSubmit = handleSubmit;
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+  }
+
+  setSubmitHandler(handler) {
+    this._handleSubmit = handler;
+  }
+
+  setEventListeners() {
+    this._popupForm.addEventListener("submit", (event) => {
+      event.preventDefault();
+      this._handleSubmit();
+      this.close();
```

Comment:
* You haven't deleted something https://github.com/rachel-naisbett/se_project_aroundtheus/pull/1/files#r1592980608 


Please, delete line 18

You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well